back-end/config/config.go

```diff
@@ -0,0 +1,44 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/viper"
+)
+
+var varTypes = [5]string{
+	"DBHost",
+	"DBName",
+	"DBPort",
+	"DBUser",
+	"DBPassword",
+}
+
+// RetrieveDBEnvVariables Finds and assigns
+// ALL key-values under config.yml to
+// env variables
+func RetrieveDBEnvVariables() {
+	viper.SetConfigName("config")
+	// Could add multiple paths if neccesary
+	viper.AddConfigPath("../../..")
+	viper.SetConfigType("yml")
+
+	// Maybe be able to use this to change
+	// the db live for faster testing.
+	// viper.WatchConfig()
+	// It also has an event handler for changes
+	// viper.OnConfigChange(func)
+
+	if err := viper.ReadInConfig(); err != nil {
+		fmt.Printf("Error reading config file, %s", err)
+	}
+
+	dbVars := viper.Sub("database")
```

Comment:
sub just accesses a sublevel like a node in the tree, and all my config settings are stored under database.